src/main/java/com/checkmarx/flow/custom/ServiceNowTracker.java

```diff
@@ -186,14 +186,19 @@ private Issue mapToIssue(Incident incident) {
     @Override
     public Issue createIssue(ScanResults.XIssue resultIssue, ScanRequest request) throws MachinaException {
         log.debug("Executing createIssue ServiceNow API call");
+        String errorMessage = "Error occurred while creating ServiceNow Issue";
+
         try {
             Incident incident = getCreateIncident(resultIssue, request);
             String query = String.format("%s%s", properties.getApiUrl(), INCIDENTS);
-            URI uri = restOperations.postForLocation(query, incident);
-            String sysId = getSysID(uri.getPath());
-            return getIncidentByIDConvertToIssue(sysId).get();
+
+            return Optional.ofNullable(restOperations.postForLocation(query, incident))
+                    .map(uri -> getSysID(uri.getPath()))
+                    .map(sysId -> getIncidentByIDConvertToIssue(sysId).get())
```

Comment:
the get can still throw null, do it in another .map

Reply:
Right. Done.